e2e/snapshot_e2e/lifecycle.py

```
@@ -262,6 +308,57 @@ def detail() -> str:
+    *,
+    allowed_reasons: set[str],
+) -> str:
+    """Asserts the invariant parts of a terminal failure vector and returns the reason.
+
+    Every failure path backfills all four conditions, `Completed` never flips
+    `True`, and `completedAt` is stamped. Where the exact reason is a genuine
+    race (e.g. the Job controller vs. the capture pipeline observing the same
+    dead workload), callers pass the allowed reason set instead of pinning one.
+    """
+    failed = condition(sj, "Failed")
+    assert failed and failed.get("status") == "True", f"Failed condition: {failed}"
```

Ruff (0.16.2)

[warning] Assertion should be broken down into multiple parts

(PT018)